packages/client/src/components/SafeOverview.js

```diff
@@ -59,34 +65,81 @@ function SafeOverview({ allBalance }) {
-                        <Svg name={type} />
-                        <label className="ml-2 pt-.5 has-text-black">
-                          {type}
-                        </label>
-                      </div>
-                      <h2 className="title is-5 has-text-white has-text-weight-normal has-text-heighlight">
-                        {balance.toLocaleString()}
-                      </h2>
+                {tokensWithPositiveBalance.map(({ type, balance }) => (
+                  <div
+                    key={type}
+                    className="has-background-secondary-purple rounded-sm p-3 m-1"
+                    style={{ flex: "2 1 auto", maxWidth: "50%" }}
+                  >
+                    <div className="is-flex is-flex-direction-row is-justify-content-flex-start mb-5">
+                      <Svg name={type} />
+                      <label className="ml-2 pt-.5 has-text-black">
+                        {type}
+                      </label>
                     </div>
-                  ))}
+                    <h2 className="title is-5 has-text-white has-text-weight-normal has-text-heighlight">
+                      {balance.toLocaleString()}
+                    </h2>
+                  </div>
+                ))}
               </div>
             )}
           </div>
         </div>
         <div
-          className="rounded-sm border-light has-shadow p-5 is-flex"
+          className="is-flex is-flex-direction-column flex-1 rounded-sm has-background-purple has-shadow p-5"
           style={{ minWidth: 375 }}
-        ></div>
+        >
+          <div className="is-flex is-flex-direction-row is-justify-content-space-between mb-5">
+            <label>NFTs</label>
+            {!!collectionsForDisplay.length && (
+              <Link to={(location) => `${location.pathname}/NFTs`}>
+                <strong className="link">View All</strong>
+              </Link>
+            )}
+          </div>
+          {!collectionsForDisplay.length && (
+            <div className="p-3 mb-1">
+              <div className="is-flex is-flex-direction-column is-justify-content-center is-align-items-center">
+                <p>You don't have any NFTs</p>
+                <button className="button column is-half mt-2 is-primary">
+                  Deposit NFTs
+                </button>
+              </div>
+            </div>
+          )}
+          {!!collectionsForDisplay.length && (
+            <div className="columns mt-1 is-multiline is-variable is-1">
+              {collectionsForDisplay.map(({ contractName, count }) => (
+                <div className="column is-half py-1" key={contractName}>
+                  <p className="has-background-tertiary-purple rounded-sm p-3 is-flex is-flex-direction-row is-justify-content-space-between">
+                    <span className="flex-3 has-text-truncate">
+                      {contractName}
```

Comment:
Name can be extracted from MetadataViews getNFTCollectionDisplay.
All NFT contracts should have it defined, so we need a script to fetch this.
Maybe even an icon can be extracted this way
```
return MetadataViews.NFTCollectionDisplay(
                        name: "Zeedz",
                        description: "Zeedz is the first play-for-purpose game where players reduce global carbon emissions by collecting plant-inspired creatures: Zeedles. They live as NFTs on an eco-friendly blockchain (Flow) and grow with the real-world weather.",
                        externalURL: MetadataViews.ExternalURL("https://play.zeedz.io"),
                        squareImage: mediaSquare,
                        bannerImage: mediaBanner,
                        socials: socials
                        )
```

Reply:
Do you mean using a async call to get the NFTCollectionDisplay?

Reply:
Yup!

Reply:
ticket created here https://linear.app/dappercollectives/issue/VSL-217